src/components/main-view/main-view.jsx

```diff
@@ -21,35 +37,161 @@ export const MainView = () => {
           };
         });
         setMovies(moviesFromApi);
-        console.log("set movies:", setMovies(moviesFromApi));
-        console.log(data);
       });
-  }, []);
+  }, [token]);
+
+  const hanldeLogout = () => {
+    localStorage.removeItem("user");
+    localStorage.removeItem("token");
+    setUser(null);
 
-  if (selectedMovie) {
-    return (
-      <MovieView
-        movie={selectedMovie}
-        onBackClicked={() => setSelectedMovie(null)}
-      />
-    );
-  }
-  console.log("SetMovies 2222: ", setMovies);
+    return <Navigate to="/login" />;
+  };
```

Comment:
A. Here I would add two functions:
`addToFavorites` & `removeFromFavorites`. Each function should tap onto the API endpoint and use the user object returned in the response.
`setUser(responseFromAPI)`

Reply:
How would I get the current movie._id variable from MovieCard